### PR TITLE
Don't force maximum stat cap in freeform mode

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
```SUMMARY: Balance "Don't force maximum stat cap in freeform mode."```

#### Purpose of change
Resolve issues with lowered (and hardcoded) maximum stats cap after #38423.

#### Describe the solution
If freeform mode, max stat points is 99, otherwise it's equals to `MAX_STAT` constant (=14).

#### Describe alternatives you've considered
Make the point after which increasing stats begins to cost double adjustable too?

#### Testing
Started game, chose freeform mode, successfully gave my character 99 strength.

#### Additional context
None.

### CODE REVIEW

**Comment by Zireael07** [comment]
Why proliferate options?

**Comment by Night-Pryanik** [comment]
> Why proliferate options?

Why not? Is that bad?

**Comment by Zireael07** [comment]
I seem to remember many people, including Kevin, complaining about too many options in the menu.

**Comment by Night-Pryanik** [comment]
I feel a one new "debug" option wouldn't hurt that much.
Well, anyway, let's see what Kevin say.

**Comment by kevingranade** [comment]
An external option would be ok, but I don't want this taking up space on the option menu.
Also this points out an issue with adjusting this number, the things that are using this number as a component of calculating costs or success chances should not be doing so.

**Comment by Night-Pryanik** [comment]
> Also this points out an issue with adjusting this number, the things that are using this number as a component of calculating costs or success chances should not be doing so.

Could you please elaborate? I don't get it.

**Comment by Night-Pryanik** [comment]
Seems like external options don't work for character generation. I got this message
изображение
when I replaced the option in `options.cpp` with a new external option in `game_balance.json`:

```
{
    "type": "EXTERNAL_OPTION",
    "name": "MAX_STAT_POINTS",
    "info": "Maximum stat points available for character generation.",
    "stype": "int",
    "value": 14
}
```

**Comment by I-am-Erk** [comment]
I would suggest that a better solution would be to have Freeform ignore the cap entirely. There's nothing wrong with having an external option for it as well, though, if that can be made to work correctly.

**Comment by Night-Pryanik** [comment]
Ok, added check for freeform mode. Maximum stat in this mode is 99.

**Comment by kevingranade** [comment]
The issue is that in iuse.cpp it references the MAX_STAT number as part of a formula, so if you adjust that value you also adjust how that formula scales, which is definitely unintended.  To be clear this is not a problem you caused, the iuse functions should not have done it in the first place.

File: src/newcharacter.cpp [hunk 1]
```diff
@@ -155,7 +154,7 @@ static matype_id choose_ma_style( const character_type type, const std::vector<m
 
 void avatar::randomize( const bool random_scenario, points_left &points, bool play_now )
 {
-
+    const int max_stat_points = points.is_freeform() ? 99 : MAX_STAT;
```
Comment (kevingranade) [inline src/newcharacter.cpp]:
```suggestion
    const int max_stat_points = points.is_freeform() ? 20 : MAX_STAT;
```

File: src/newcharacter.cpp [hunk 2]
```diff
@@ -735,6 +734,8 @@ tab_direction set_points( const catacurses::window &w, avatar &, points_left &po
 
 tab_direction set_stats( const catacurses::window &w, avatar &u, points_left &points )
 {
+    const int max_stat_points = points.is_freeform() ? 99 : MAX_STAT;
```
Comment (kevingranade) [inline src/newcharacter.cpp]:
```suggestion
    const int max_stat_points = points.is_freeform() ? 20 : MAX_STAT;
```

**Pull request merged**